contrib/gcppubsub/samples/README.md

```diff
@@ -12,112 +20,110 @@
+       check this key into source control!**
+
+       ```shell
+       gcloud iam service-accounts keys create knative-source.json \
+         --iam-account=knative-source@$PROJECT_ID.iam.gserviceaccount.com
+       ```
+
+    1. Create two secrets on the kubernetes cluster with the downloaded key:
+                                                                                                      
+       ```shell
+       # Note that the first secret may already have been created when installing
+       # Knative Eventing. The following command will overwrite it. If you don't
+       # want to overwrite it, then skip this command.
+       kubectl -n knative-sources create secret generic gcppubsub-source-key --from-file=key.json=knative-source.json --dry-run -o yaml | kubectl apply --filename -
+
+       # The second secret should not already exist, so just try to create it.
+       kubectl -n default create secret generic google-cloud-key --from-file=key.json=knative-source.json
+       ```
+
+       `gcppubsub-source-key` and `key.json` are pre-configured values in the
+       `controller-manager` StatefulSet which manages your Eventing sources.
+
+       `google-cloud-key` and `key.json` are pre-configured values in
+       [`gcp-pubsub-source.yaml`](./gcp-pubsub-source.yaml). If you choose to
+       create a second account, this account only needs
+       `roles/pubsub.Subscriber`.
 
-        - Note that you can change the secret's name and the secret's key, but
-          will need to modify `default-gcppubsub.yaml` in a later step with the
-          updated values (they are environment variables on the `StatefulSet`).
 
-   - The Receive Adapter's Service Account.
-
-     1. Determine the Service Account to use, or create a new one.
-     1. Give that Service Account the 'Pub/Sub Subscriber' role on your GCP
-        project.
-     1. Download a new JSON private key for that Service Account.
-     1. Create a secret for the downloaded key in the namespace that the
-        `Source` will be created in:
-
-        ```shell
-        kubectl -n default create secret generic google-cloud-key --from-file=key.json=PATH_TO_KEY_FILE.json
-        ```
-
-        - Note that you can change the secret's name and the secret's key, but
-          will need to modify `gcp-pubsub-source.yaml`'s `spec.gcpCredsSecret`
-          in a later step with the updated values.
+### Deployment
 
 1. Create a GCP PubSub Topic. Replace `TOPIC-NAME` with your desired topic name.
 
    ```shell
    gcloud pubsub topics create TOPIC-NAME
    ```
 
-### Deployment
-
-1. Deploy the `GcpPubSubSource` controller as part of eventing-source's
-   controller.
-
-   ```shell
-   ko -n default apply -f config/default-gcppubsub.yaml
-   ```
-
-   - Note that if the `Source` Service Account secret is in a non-default
-     location, you will need to update the YAML first.
-
 1. Replace the place holders in `gcp-pubsub-source.yaml`.
```

Comment:
I recommend doing this as setting environment variables, then using sed on the yaml. So that people don't have to manipulate the file directly.

```shell
export TOPIC_NAME=laconia

sed -e "s/MY_GCP_PROJECT/$PROJECT_ID/g" -e "s/TOPIC_NAME/$TOPIC_NAME/g" gcp-pubsub-source.yaml | kubectl apply -
```

Reply:
Done, thanks!

Reply:
Export PROJECT_ID and TOPIC_NAME.

Reply:
Done -- thanks for prodding me to find a copy-pasteable way to get the project ID.